gateway/mw_version_check.go

```diff
@@ -68,22 +68,45 @@ func (v *VersionCheck) ProcessRequest(w http.ResponseWriter, r *http.Request, _
 
 	ctxSetSpanAttributes(r, v.Name(), otel.APIVersionAttribute(targetVersion))
 
-	if v.Spec.VersionDefinition.Enabled && targetVersion != apidef.Self && targetVersion != v.Spec.VersionDefinition.Name {
+	isBase := func(vName string) bool {
+		return vName == apidef.Self || vName == v.Spec.VersionDefinition.Name
+	}
+
+	if v.Spec.VersionDefinition.Enabled && !isBase(targetVersion) {
 		if targetVersion == "" {
 			return errors.New(string(VersionNotFound)), http.StatusForbidden
 		}
 
 		subVersionID := v.Spec.VersionDefinition.Versions[targetVersion]
 		handler, _, found := v.Gw.findInternalHttpHandlerByNameOrID(subVersionID)
 		if !found {
-			return errors.New(string(VersionDoesNotExist)), http.StatusNotFound
+			if !v.Spec.VersionDefinition.FallbackToDefault {
+				return errors.New(string(VersionDoesNotExist)), http.StatusNotFound
+			}
+
+			if isBase(v.Spec.VersionDefinition.Default) {
```

Comment:
can we invert this condition to have the statements after this block so that we can avoid the use of goto?

Reply:
It is to prevent lookup

Reply:
Can you please write in more dtail?